rotkehlchen/db/dbhandler.py

```diff
@@ -342,6 +342,19 @@ def connect(self, password: str) -> None:
             self.conn.executescript(script)
         self.conn.execute('PRAGMA foreign_keys=ON')
 
+    def change_password(self, new_password: str) -> None:
+        """Changes the password for the currently logged in user
+        """
+        self.conn.text_factory = str
+        new_password_for_sqlcipher = _protect_password_sqlcipher(new_password)
+        self.conn.executescript(f'PRAGMA rekey="{new_password_for_sqlcipher}";')
```

Comment:
You are not catching errors in the first case. Can you instead do the following?

```python
script = f'PRAGMA rekey="{new_password_for_sqlcipher}";'
if self.sqlcipher_version == 3:
    script +=  'PRAGMA kdf_iter={KDF_ITER};';
try:
    self.conn.executescript(script)
except sqlcipher.OperationalError as e:  # pylint: disable=no-member
     log.error(f'At change password could not re-key the open database: {str(e)}')
```

By looking at the current `connect()` code I see that a similar pattern is employed. You can also replace it there. By the way not sure if it makes sense to account for sqlcipher version 3. We ship sqlcipher with the binary and it's version 4 now. This is only if someone has it built manually and his system has version 3.

Reply:
Think I got it but for some reason pytest is choking when running more than 1 test at the same time (something about the db being in use). So I will upload with next commit and check with gh actions.